gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/KubernetesWatchTask.java

```diff
@@ -0,0 +1,117 @@
+        WatcherContext context = createWatcherContext();
+
+        WatcherManager.watch(resolvedImages, resources, context);
+      } catch (KubernetesClientException kubernetesClientException) {
+        KubernetesResourceUtil.handleKubernetesClientException(kubernetesClientException, kitLogger);
+      } catch (Exception ioException) {
+        throw new IllegalStateException("An error has occurred while while trying to watch the resources", ioException);
+      }
+    }
+  }
+
+  private WatcherContext createWatcherContext() throws IOException {
+    WatchContext watchContext = jKubeServiceHub.getDockerServiceHub() != null ? getWatchContext() : null;
+    return WatcherContext.builder()
+        .buildContext(jKubeServiceHub.getConfiguration())
+        .watchContext(watchContext)
+        .config(extractWatcherConfig())
+        .logger(kitLogger)
+        .newPodLogger(createLogger("[[C]][NEW][[C]] "))
+        .oldPodLogger(createLogger("[[R]][OLD][[R]] "))
+        .useProjectClasspath(kubernetesExtension.getUseProjectClassPathOrDefault())
+        .jKubeServiceHub(jKubeServiceHub)
+        .build();
+  }
+
+  private ProcessorConfig extractWatcherConfig() {
+    try {
+      return ProfileUtil.blendProfileWithConfiguration(ProfileUtil.WATCHER_CONFIG, kubernetesExtension.getProfileOrNull(), ResourceUtil.getFinalResourceDir(kubernetesExtension.getResourceSourceDirectoryOrDefault(), kubernetesExtension.getResourceEnvironmentOrNull()), kubernetesExtension.watcher);
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Cannot extract watcher config: " + e, e);
+    }
+  }
+
+  private WatchContext getWatchContext() throws IOException {
+    final ServiceHub hub = jKubeServiceHub.getDockerServiceHub();
+    return WatchContext.builder()
+        .watchInterval(kubernetesExtension.getWatchIntervalOrDefault())
+        .watchMode(kubernetesExtension.getWatchModeOrDefault())
+        .watchPostExec(kubernetesExtension.getWatchPostExecOrNull())
+        .autoCreateCustomNetworks(kubernetesExtension.getWatchAutoCreateCustomNetworksOrDefault())
+        .keepContainer(kubernetesExtension.getWatchKeepContainerOrDefault())
+        .keepRunning(kubernetesExtension.getWatchKeepRunningOrDefault())
+        .removeVolumes(kubernetesExtension.getWatchRemoveVolumesOrDefault())
+        .containerNamePattern(kubernetesExtension.getWatchContainerNamePatternOrDefault())
+        .buildTimestamp(getBuildTimestamp(null, null, kubernetesExtension.javaProject.getBuildDirectory().getAbsolutePath(), DOCKER_BUILD_TIMESTAMP))
+        .gavLabel(new GavLabel(kubernetesExtension.javaProject.getGroupId(), kubernetesExtension.javaProject.getArtifactId(), kubernetesExtension.javaProject.getVersion()))
+        .buildContext(jKubeServiceHub.getConfiguration())
+        .follow(kubernetesExtension.getWatchFollowOrDefault())
+        .showLogs(kubernetesExtension.getWatchShowLogsOrNull())
+        .serviceHubFactory(new ServiceHubFactory())
```

Comment:
There's something odd here.

This is ultimately used to retrieve the logOutputSpecFactory. Howeverm since this is a **new** factory instance, the service hub is never created and the logOutputSpecFactory is never initialized.

In this case, maybe we need to access to the original [ServiceHubFactory](https://github.com/eclipse/jkube/blob/cb53c045d3946ed749487797ca88c7db460e4b93/gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/TaskUtil.java#L67).

Maybe the best approach is just to modify the WatchContext and provide the applicable field.